packages/http/src/types.ts

```diff
@@ -90,16 +102,37 @@ export class ProblemJsonError extends Error {
     return error;
   }
 
+  public static asValidationIssue(error: Error & { detail: IPrismDiagnostic[]; status?: number }): any {
+    return {
+      ...sharedErrorProps(error),
+      detail:
+        'Your request body is not valid and no HTTP validation response was found in the spec, so Prism is generating this error for you.',
+      validation: error.detail.map(detail => {
+        const { path, code, message } = detail;
+
+        return {
+          location: path,
+          severity: DiagnosticSeverity[detail.severity],
+          code,
+          message,
+        };
+      }),
+    };
+  }
+
   public static fromPlainError(error: Error & { detail?: string; status?: number }): ProblemJson {
     return {
-      type: error.name && error.name !== 'Error' ? error.name : 'https://stoplight.io/prism/errors#UNKNOWN',
-      title: error.message,
-      status: error.status || 500,
+      ...sharedErrorProps(error),
       detail: error.detail || '',
     };
   }
 
-  constructor(readonly name: string, readonly message: string, readonly status: number, readonly detail: string) {
+  constructor(
+    readonly name: string,
+    readonly message: string,
+    readonly status: number,
+    readonly detail: string | IPrismDiagnostic[],
```

Comment:
Is this correct though? According to the code I saw so far it's always a string — and then maybe there might be an additional `validation` property; that one is an `IPrismDiagnostic` one.

Reply:
https://github.com/stoplightio/prism/blob/47d4cad1151b97974e754b0e36de57347fed217e/packages/http/src/mocker/HttpMocker.ts#L62
and then `fromTemplate` does `const error = new ProblemJsonError(...)`